packages/CopyInput/src/CopyInput.js

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Input from "@paprika/input";
+import Popover from "@paprika/popover";
+import Button from "@paprika/button";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import extractChildrenProps from "@paprika/helpers/lib/extractChildrenProps";
+import CopyIcon from "@paprika/icon/lib/ArrowDown"; // TODO: Replace with actual icon when ready
+import CopyInputInputPropsCollector from "./CopyInputInputPropsCollector";
+import * as sc from "./CopyInput.styles";
+
+function CopyInput(props) {
+  const { defaultValue, children, ...moreProps } = props;
+  const I18n = useI18n();
+  const extendedInputProps = extractChildrenProps(children, CopyInputInputPropsCollector);
+
+  const [value, setValue] = React.useState(defaultValue);
+  const [isClickedTooltipOpen, setIsClickedTooltipOpen] = React.useState(false);
+
+  function handleButtonClick() {
+    navigator.clipboard.writeText(value);
+    setIsClickedTooltipOpen(true);
+    setTimeout(() => setIsClickedTooltipOpen(false), 5000);
```

Comment:
Maybe you would want to move this into a sideEffect? so can be clean after the component is removed?

Let's suppose that this component is used in a sidepanel, the user clicks the button to copy but then close the panel.

What will happens is that the setTimeout will execute once the side panel is not there and will try to `setIsClickedTooltipOpen(false)` in a component that doesn't exist creating a minor memory leak. 


you could do something like this:

```js
// inside your component
const refSetTimeout = React.useRef(null);

React.useEffect(() => {
  if(isClickedTooltipOpen) {
     refSetTimeout.current = setTimeout(()=>{
     refSetTimeout.current = null
     ... // more stuff
     }, 5000)
  }

   return () => {
	if (refSetTimeout.current) {
       clearTimeOut(refSetTimeout.current)
    }
   }
}, [isClickedTooltipOpen])
```

Reply:
^ the return function in the effect is the cleaning function that will be executed before the component will be unmounted.